### PR TITLE
#1016 Delegated constructor call in external class is not allowed

### LINKED ISSUE #1016: Delegated constructor call in external class is not allowed
Sample of invalid generated code:
```
public open abstract external class AbortSignal : EventTarget() {
  public val aborted: Boolean?
}
```

### LINKED ISSUE #1018: Expect classes cannot initialize supertypes
Sample incorrectly generated code:
```
public expect open abstract class AbortSignal : EventTarget() {
  public open val aborted: Boolean?
}
```
Error: "Expected classes cannot initialize supertypes"

### PR DESCRIPTION
**Pull request by NiematojakTomasz**
Fixes #1016
Fixes #1018

### CODE REVIEW

**Comment by Egorand** [comment]
Can you please add tests for the changes? Test cases that should be covered:

- `external object`
- `external class`
- Do we cover interfaces?

**Comment by NiematojakTomasz** [comment]
@Egorand Does not affect interfaces, as interfaces do not have constructors.

**Comment by NiematojakTomasz** [comment]
@Egorand I'll add tests tomorrow probably. Should I also add some validation - code block defined for parent class constructor in external class?

**Comment by Egorand** [comment]
Yeah, that sounds like something we'd want to validate, thanks!

**Comment by NiematojakTomasz** [comment]
@Egorand Ready : )

File: kotlinpoet/src/test/java/com/squareup/kotlinpoet/DelegatedConstructorCallTest.kt [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        |public class Test {
+        |  public constructor(foo: kotlin.String) : this()
+        |}
+        |""".trimMargin()
+    )
+  }
+
+  @Test
+  fun notAllowedInExternalClass() {
+    val builder = TypeSpec.classBuilder("Test")
```
Comment (Egorand) [inline kotlinpoet/src/test/java/com/squareup/kotlinpoet/DelegatedConstructorCallTest.kt:122]: nit: the code could look a bit less verbose if you used method chaining:
```kotlin
val builder = TypeSpec.classBuilder("Test")
  .addModifiers(KModifier.EXTERNAL)
  .superclass(ClassName("testpackage", "TestSuper"))
  .addSuperclassConstructorParameter("anything")
```
As is is fine though, so not critical to fix.

**Review by Egorand — APPROVED** [review]
Looks good, thanks! Make sure you've signed our [CLA](https://spreadsheets.google.com/spreadsheet/viewform?formkey=dDViT2xzUHAwRkI3X3k5Z0lQM091OGc6MQ&ndplr=1) and we can merge the PR.

File: kotlinpoet/src/test/java/com/squareup/kotlinpoet/DelegatedConstructorCallTest.kt (hunk 1 above)
Comment (NiematojakTomasz) [reply kotlinpoet/src/test/java/com/squareup/kotlinpoet/DelegatedConstructorCallTest.kt:122]: @Egorand Done. CLA also signed.

**Pull request merged**